include/vinecopulib/vinecop/implementation/class.ipp

```diff
@@ -630,20 +645,80 @@ Vinecop::get_threshold() const
   return threshold_;
 }
 
+//! @brief sets variable types.
+//! @param var_types a vector of size two specifying the types of the variables,
+//!   e.g., `{"c", "d"}` means first varible continuous, second discrete.
+inline void
+Vinecop::set_var_types(const std::vector<std::string>& var_types)
+{
+  std::stringstream msg;
+  if (var_types.size() > d_) {
+    msg << "more var_types (" << var_types.size() << ")"
+        << "than variables (" << d_ << ")-" << std::endl;
+  }
+  for (auto t : var_types) {
+    if (!tools_stl::is_member(t, { "c", "d" })) {
+      msg << "variable type must be 'c' or 'd'." << std::endl;
```

Comment:
So the error message could be a list of `"variable type must be 'c' or 'd'."` statements one after the other? Shouldn't we write only once? Or let the user know which component is problematic?

Reply:
Agree